Source/WebCore/dom/TrustedTypeUtils.cpp

```diff
@@ -0,0 +1,162 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "TrustedTypeUtils.h"
+
+#include "ContentSecurityPolicy.h"
+#include "Document.h"
+#include "JSDOMExceptionHandling.h"
+#include "TrustedTypePolicy.h"
+#include "TrustedTypePolicyFactory.h"
+#include "WindowOrWorkerGlobalScopeTrustedTypes.h"
+#include "WorkerGlobalScope.h"
+#include <JavaScriptCore/HeapInlines.h>
+#include <JavaScriptCore/JSCInlines.h>
+#include <JavaScriptCore/JSCJSValueInlines.h>
+#include <JavaScriptCore/JSCast.h>
+
+namespace WebCore {
+using namespace JSC;
+
+struct TrustedTypeVisitor {
+    String operator()(std::nullptr_t)
+    {
+        return nullString();
+    }
+    String operator()(Exception)
+    {
+        return nullString();
+    }
+    String operator()(Ref<TrustedHTML> value)
+    {
+        return value->toString();
+    }
+    String operator()(Ref<TrustedScript> value)
+    {
+        return value->toString();
+    }
+    String operator()(Ref<TrustedScriptURL> value)
+    {
+        return value->toString();
+    }
+};
+
+String trustedTypeToString(const TrustedType trustedType)
+{
+    switch (trustedType) {
+    case TrustedType::TrustedHTML:
+        return "TrustedHTML"_s;
+    case TrustedType::TrustedScript:
+        return "TrustedScript"_s;
+    case TrustedType::TrustedScriptURL:
+        return "TrustedScriptURL"_s;
+    default:
+        ASSERT_NOT_REACHED();
+        return nullString();
+    }
+}
+
+// https://w3c.github.io/trusted-types/dist/spec/#process-value-with-a-default-policy-algorithm
+std::variant<std::nullptr_t, Exception, Ref<TrustedHTML>, Ref<TrustedScript>, Ref<TrustedScriptURL>> processValueWithDefaultPolicy(ScriptExecutionContext& scriptExecutionContext, TrustedType expectedType, const String& input, const String& sink)
+{
+    RefPtr<TrustedTypePolicy> protectedPolicy;
+    if (RefPtr document = dynamicDowncast<Document>(scriptExecutionContext)) {
+        if (auto window = document->domWindow()) {
+            auto trustedTypesFactory = WindowOrWorkerGlobalScopeTrustedTypes::trustedTypes(*window);
+            protectedPolicy = trustedTypesFactory->defaultPolicy();
+        }
+    } else if (RefPtr workerGlobalScope = dynamicDowncast<WorkerGlobalScope>(scriptExecutionContext)) {
+        auto trustedTypesFactory = WindowOrWorkerGlobalScopeTrustedTypes::trustedTypes(*workerGlobalScope);
+        protectedPolicy = trustedTypesFactory->defaultPolicy();
+    }
+
+    if (!protectedPolicy)
+        return nullptr;
+
+    VM& vm = scriptExecutionContext.vm();
+
+    auto jsExpectedType = JSC::jsString(vm, trustedTypeToString(expectedType));
+    JSC::Strong<JSC::Unknown> strongExpectedType(vm, jsExpectedType);
+    auto jsSink = JSC::jsString(vm, sink);
+    JSC::Strong<JSC::Unknown> strongSink(vm, jsSink);
+    FixedVector<JSC::Strong<JSC::Unknown>> arguments({ strongExpectedType, strongSink });
```

Comment:
I think we can write this:
```
FixedVector<JSC::Strong<JSC::Unknown>> arguments({ { vm, jsExpectedType }, { vm, jsSink } });
```
And get rid of the `strongExpectedType` and `strongSink` local variables.